tools/generator/src/Extensions/XCScheme+Extensions.swift

```diff
@@ -143,6 +155,23 @@ echo "$BAZEL_LABEL,$BAZEL_TARGET_ID" >> "$SCHEME_TARGET_IDS_FILE"
             environmentBuildable: buildableReference
         )
     }
+    
+    /// Symlinks `$DEVELOPER_DIR/Toolchains/XcodeDefault.xctoolchain/usr/lib` to `$(BAZEL_INTEGRATION_DIR)/lib` so that Xcode can copy sanitizers' dylibs.
+    static func symlinkDefaultToolchainUsrLibDirectory(
+        buildableReference: XCScheme.BuildableReference
+    ) -> XCScheme.ExecutionAction {
+        return .init(
+            scriptText: #"""
+if [ "${ENABLE_THREAD_SANITIZER:-}" == "YES" ]; then
+    dest="$BAZEL_INTEGRATION_DIR/lib"
+    DEV_LIB_DIR="$SYSTEM_DEVELOPER_DIR/Toolchains/XcodeDefault.xctoolchain/usr/lib"
+    ln -sF "$DEV_LIB_DIR" "$dest"
+fi
+"""#,
```

Comment:
My other comment seems to have gone away: but I think this should be in `bazel-build.sh` instead of a scheme script. Also, we can check `BAZEL_BUILD_MODE` and only do this when `!= "xcode"`. Also also, since we have the info available we can support the custom toolchain correctly here (but shouldn't since it wouldn't match the stub currently, so add a comment to that effect).